### PR TITLE
[Frontend] Make links in pipeline description clickable

### PR DESCRIPTION
**Pull request by Bobgy**
Improves usability of pipeline description.

~demo: https://drive.google.com/file/d/1Z-W6yeIQNVxXF1J0KwKcY1UDPVR_8Tmo/view~
new demo (changed implementation to markdown link):
localhost_3000_ (2)

Changes:
* Pipeline list description is rendered as inline markdown
* Pipeline detail description is rendered as normal markdown
* Updated markdown viewer to also use react-to-jsx because it is smaller in bundle size and better integrated with react. (It doesn't quite make sense we use two markdown parsers.)
* Updated sample README using new markdown syntax for links

/area front-end
/kind feature
/cc @jessiezcc

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2376)

### CODE REVIEW

**Comment by Bobgy** [comment]
@jessiezcc and @gaoning777, I updated demo screenshot (its data is purely faked to show utility).
Do you have any further suggestions on the wording?

Changes:
1. This now uses markdown url syntax.
2. Updated sample description to put permission requirements at first word.

**Comment by gaoning777** [comment]
> @jessiezcc and @gaoning777, I updated demo screenshot (its data is purely faked to show utility).
> Do you have any further suggestions on the wording?
> 
> Changes:
> 
> 1. This now uses markdown url syntax.
> 2. Updated sample description to put permission requirements at first word.

This is great. BTW, is the link fonts color rendered as purple because it's been clicked? Normally links are blue.

**Comment by Bobgy** [comment]
> This is great. BTW, is the link fonts color rendered as purple because it's been clicked? Normally links are blue.

Yes, it's been clicked

**Comment by Bobgy** [comment]
@neuromage @IronPan needs your review too, because I touched backend/ samples

File: backend/src/apiserver/config/sample_config.json [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+    "file": "/samples/core/exit_handler/exit_handler.py.tar.gz"
   },
   {
-    "name":"[Sample] ML - TFX - Taxi Tip Prediction Model Trainer",
-    "description":"Example pipeline that does classification with model analysis based on a public tax cab BigQuery dataset. GCS permission is required. For guideline of setting up permission after deployment, please see https://github.com/kubeflow/pipelines/blob/master/manifests/gcp_marketplace/guide.md#gcp-service-account-credentials. For source code, refer to https://github.com/kubeflow/pipelines/tree/master/samples/contrib/parameterized_tfx_oss",
-    "file":"/samples/contrib/parameterized_tfx_oss/parameterized_tfx_oss.tar.gz"
+    "name": "[Sample] ML - TFX - Taxi Tip Prediction Model Trainer",
+    "description": "[Permission requirements](https://github.com/kubeflow/pipelines/blob/master/manifests/gcp_marketplace/guide.md#gcp-service-account-credentials). Example pipeline that does classification with model analysis based on a public tax cab BigQuery dataset. GCS permission is required. [source code](https://github.com/kubeflow/pipelines/tree/master/samples/contrib/parameterized_tfx_oss)",
```
Comment (jessiezcc) [inline backend/src/apiserver/config/sample_config.json]: This should point to TFX sample README file which explains service account permission settings. The link you have here is general service account creation, but specific to this sample. The TFX sample README references this link, but with additional service account role instructions.

File: backend/src/apiserver/config/sample_config.json [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
+    "name": "[Sample] Basic - Sequential execution",
+    "description": "A pipeline with two sequential steps. [source code](https://github.com/kubeflow/pipelines/blob/master/samples/core/sequential/sequential.py)",
+    "file": "/samples/core/sequential/sequential.py.tar.gz"
   },
   {
-    "name":"[Sample] Basic - Parallel execution",
-    "description":"A pipeline that downloads two messages in parallel and prints the concatenated result. For source code, refer to https://github.com/kubeflow/pipelines/blob/master/samples/core/parallel_join/parallel_join.py",
-    "file":"/samples/core/parallel_join/parallel_join.py.tar.gz"
+    "name": "[Sample] Basic - Parallel execution",
+    "description": "A pipeline that downloads two messages in parallel and prints the concatenated result. [source code](https://github.com/kubeflow/pipelines/blob/master/samples/core/parallel_join/parallel_join.py)",
```
Comment (jessiezcc) [inline backend/src/apiserver/config/sample_config.json]: I believe XGBoost sample also needs Permission Requirement link. @numerology who owns this sample

Comment (numerology) [reply backend/src/apiserver/config/sample_config.json]: Thanks for bringing this up. Yes it requires dataproc permission.

Comment (numerology) [reply backend/src/apiserver/config/sample_config.json]: XGboost sample needs storage.admin permission as well

File: backend/src/apiserver/config/sample_config.json (hunk 1 above)
Comment (numerology) [reply backend/src/apiserver/config/sample_config.json]: Thanks for pointing this out. Sent #2422 to fix.

**Comment by numerology** [comment]
Thanks for doing this @Bobgy

File: backend/src/apiserver/config/sample_config.json (hunk 2 above)
Comment (jessiezcc) [reply backend/src/apiserver/config/sample_config.json]: Is this described in XGBoost README file? @gaoning777 has a PR for it, I remember.
Yuan, you can just point to README file.

Comment (numerology) [reply backend/src/apiserver/config/sample_config.json]: Yes @gaoning777 made #2359 for this.

File: backend/src/apiserver/config/sample_config.json (hunk 1 above)
Comment (Bobgy) [reply backend/src/apiserver/config/sample_config.json]: SG, thanks for the fix!

Comment (Bobgy) [reply backend/src/apiserver/config/sample_config.json]: Updated link

File: backend/src/apiserver/config/sample_config.json (hunk 2 above)
Comment (Bobgy) [reply backend/src/apiserver/config/sample_config.json]: Updated link

**Comment by Bobgy** [comment]
/approve
for frontend part

@IronPan Can you approve for backend?

**Comment by Bobgy** [comment]
Made some minor styling changes:
1. Use our theme color
2. Hide underline initially and show it on hover
demo: https://drive.google.com/file/d/1TVXJovghgUOTSQIO8vBkqo6LVOEhAC6H/view

**Pull request merged**